File: src/main/java/funkies/errors/UserError.java
package funkies.errors;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

//100-199
@Getter
@RequiredArgsConstructor
public enum UserError implements SystemErrorCode {
    ACCESS_DENIED(100, "Access is denied for this user");

    final int code;

    final String message;

}

File: src/main/kotlin/kfunkies/errors/UserError.kt
package kfunkies.errors

enum class UserError(
    private val code: Int,
    private val message: String
) : SystemErrorCode {
    ACCESS_DENIED(code = 100, message = "Access is denied for this user");

    override fun getCode() = code
    override fun id() = name
    override fun getMessage() = message
}